### DISCUSSION TITLE
Charge Used data accuracy

### DISCUSSION THREAD

**Discussion by ghost** [teslamate-org/teslamate#1919] [Q&A] [answered]
I did complete the first charging sessions and compared the used amount against the bills from the providers.

It looks like the 'Used' amount is slightly lower than the amount charged in reality:

<img width="1361" alt="Bildschirmfoto 2021-08-21 um 10 41 36" src="https://user-images.githubusercontent.com/3201804/130317379-67622db4-2d5c-4765-819d-bdc9931866b6.png">

![IMG_BA6147FE72A1-1](https://user-images.githubusercontent.com/3201804/130317388-5ad694a2-b070-4bc9-ba86-f54d630c8f9a.jpeg)

So, what is the technical reason here? I know that the outlets are well-calibrated.

Is there a way to adjust this 'offset' or edit the data manually afterwards?

**Comment by eliluong** [comment]
You should be able to use something like pgAdmin to directly edit the database information for the charging sessions.

**Comment by adriankumpf** [comment, marked as answer]
The "energy used" is _estimated_ by TeslaMate based on the current/voltage provided by the vehicle via the Tesla API. Any losses of the charger etc. are not taken into account.

**Comment by k3it** [comment]
My wall connector reports higher usage per charge session than teslamate  due to some losses at 48A charging current which the car cannot measure and report.   

There are further losses in the wiring between the wall connector and the electric meter (in my case 68 feet of AWG 6 copper wire).  

The total loss in the connector and the house wiring is approx 5%.   

To make teslamate indicate the correct cost I simply increased the price per kW by 5% for my home location.    

When setting the cost don't forget to account for distribution and various utility surcharges, minus the TOU discount if applicable.   The final cost for me as of today is 14.86 cents per kW delivered to the vehicle.    compare that to the utility advertised "comparison" EV charging rate of 8.6 cents per kW.  

so the reality is slightly less rosy than the marketing😛
